File: src/budget_api/models/partners.py
"""Partner pydantic models — Partner, PartnerCreate, PartnerUpdate, PartnerList."""

from __future__ import annotations

from pydantic import BaseModel


class Partner(BaseModel):
    """Stored partner — id slugified from label."""

    id: str
    label: str


class PartnerCreate(BaseModel):
    """POST /api/partners body — label validated manually in router."""

    label: str


class PartnerUpdate(BaseModel):
    """PUT /api/partners/{id} body — label validated manually in router."""

    label: str


class PartnerList(BaseModel):
    """GET /api/partners response."""

    partners: list[Partner]

File: client/src/api/partners.ts
import { apiDelete, apiGet, apiPost, apiPut } from "./client";
import type { Partner, PartnerCreate, PartnerList, PartnerUpdate } from "@/types/api";

// List partners.
export function listPartners(): Promise<PartnerList> {
  return apiGet<PartnerList>("/api/partners");
}

// Create partner.
export function createPartner(label: string): Promise<Partner> {
  const body: PartnerCreate = { label };
  return apiPost<Partner>("/api/partners", body);
}

// Update partner label.
export function updatePartner(id: string, label: string): Promise<Partner> {
  const body: PartnerUpdate = { label };
  return apiPut<Partner>(`/api/partners/${encodeURIComponent(id)}`, body);
}

// Delete partner.
export function deletePartner(id: string): Promise<void> {
  return apiDelete<void>(`/api/partners/${encodeURIComponent(id)}`);
}